### DISCUSSION TITLE
Add support for Instagram Recipe Scraping

### DISCUSSION THREAD

**Discussion by walk0080** [mealie-recipes/mealie#3846] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Instagram recipe scraping is perhaps only usable via a screen capture of a recipe and using iOS Shortcut and OCR.

Would be great if Mealie supported scraping Instagram recipes when the recipe is in the description text.
e.g. recipe link: https://www.instagram.com/p/C6ovhr4ocV0/

### Please provide a concise description of the feature that would resolve your issue.

Add scraping support for Instagram recipes.

### Please consider and list out some caveats or tradeoffs made in your design decision

N/A

### Additional Information

- [X] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by JoTec2002** [comment]
I wanted to have a similar feature. Mealie already offers some integration with OpenAI/Ollama, but I wasn't fully satisfied with the results.

So, I created a small Docker container that explicitly scrapes Instagram posts using an Instagram scraper and allows Mealie to create a new recipe based on the video description. Afterward, it uploads the Instagram image and video directly to Mealie.

I hope this might help someone: https://github.com/JoTec2002/InstagramToMealie/
It creates a small Website were you can input the Instagram URL. I have it deployed in the same Docker-Compose file as Mealie.

I hope it’s okay to share this here—this isn’t for marketing; I just thought it might be useful to someone else!

**Reply by stusemple** [reply]
wow this is amazing, thanks so much for sharing. Can you help me with the instagram session file. I am struggling to understand how to generate one. thanks

**Reply by Kuchenpirat** [reply]
This is absolutely more than OK to share here! 

We love to see community projects interacting with mealie in new and creative ways 🚀😊

**Reply by JoTec2002** [reply]
@stusemple i tried explaining it further in the discussion you opened in the other repo.

Linking it here so others could find it more easily: https://github.com/JoTec2002/InstagramToMealie/discussions/1
